test/unit/data-generator.js

```diff
@@ -0,0 +1,250 @@
+    runner.emit('suite', input.rootSuite);
+    runner.emit('test end', test);
+    runner.emit('suite end', input.rootSuite);
+    runner.emit('end');
+  });
+
+  it('should generate a Test for each of Mocha\'s passed it', function(done) {
+    var randomNumber = getRandomNumber();
+
+    generate(runner, function(data) {
+      var tests = generateTests(randomNumber, 'passed'),
+          expected = [objectAssign({}, output.rootSuite, {tests: tests})];
+
+      expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite',input.rootSuite);
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('test end', input.passTest);
+    }
+
+    runner.emit('suite end', input.rootSuite);
+    runner.emit('end');
+  });
+
+  it('should generate a Test for each of Mocha\'s failed it', function(done) {
+    var randomNumber = getRandomNumber();
+
+    generate(runner, function(data) {
+      var tests = generateTests(randomNumber, 'failed'),
+          expected = [objectAssign({}, output.rootSuite, {tests: tests})];
+
+      expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite',input.rootSuite);
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('test end', input.failTest);
+    }
+
+    runner.emit('suite end', input.rootSuite);
+    runner.emit('end');
+  });
+
+  it('should generate a Suite for each of Mocha\'s describe', function(done) {
+    var randomNumber = getRandomNumber();
+
+    generate(runner, function(data) {
+      var expected = generateSuites(randomNumber, 'global');
+
+      expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite', input.rootSuite);
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('suite', input.suite);
+      runner.emit('suite end', input.suite);
+    };
+
+    runner.emit('suite end', input.rootSuite);
+    runner.emit('end');
+  });
+
+  it('should indent nested suites properly', function(done) {
+    var randomNumber = getRandomNumber();
+
+    generate(runner, function(data) {
+      var expected = generateSuites(randomNumber, 'nested');
+
+      for (var i = 0; i <= randomNumber; i++) {
+        expect(data[i].indent).to.be.equal(expected[i].indent);
+      }
+
+      done();
+    });
+
+    runner.emit('suite', input.rootSuite);
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('suite', input.suite);
+    }
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('suite end', input.suite);
+    }
+
+    runner.emit('suite end', input.rootSuite);
+    runner.emit('end');
+  });
+
+  it('should maintain the order in the tests are executed', function(done) {
+    var n = getRandomNumber(),
+        // Each value is the number of tests of the suite i.
+        randomNumbers = generateArrayOfRandomNumbers(n);
+
+    generate(runner, function(data) {
+      var expected = generateSuites(n, 'global');
+
+      for (var i = 1; i <= n; i++) {
```

Comment:
Because the `randomNumbers` array contains, only n numbers and I have `rootSuite + n Suites`. And I left the rootSuite void. I will put some tests into the rootSuite too :smile: